server/cluster/coordinator.go

```diff
@@ -157,6 +162,33 @@ func (c *coordinator) patrolRegions() {
 	}
 }
 
+// checkSuspectKeyRanges would pop one suspect key range group
+// The regions of new version key range and old version key range would be placed into
+// the suspect regions map
+func (c *coordinator) checkSuspectKeyRanges() {
+	keyRangeGroup, success := c.cluster.PopOneSuspectKeyRange()
+	if !success {
+		return
+	}
+	regionMap := map[uint64]struct{}{}
+	f := func(regions []*core.RegionInfo) {
+		for _, region := range regions {
+			regionMap[region.GetID()] = struct{}{}
+		}
+	}
+	newKeyRegions := c.cluster.ScanRegions(keyRangeGroup[0], keyRangeGroup[1], patrolScanRegionLimit)
+	f(newKeyRegions)
+	if !bytes.Equal(keyRangeGroup[0], keyRangeGroup[2]) || !bytes.Equal(keyRangeGroup[1], keyRangeGroup[3]) {
+		oldKeyRegions := c.cluster.ScanRegions(keyRangeGroup[2], keyRangeGroup[3], patrolScanRegionLimit)
+		f(oldKeyRegions)
+	}
+	var regionIDList []uint64
```

Comment:
How about initializing it with the capacity?

Reply:
updated.